modules/priceFloors.js

```diff
@@ -295,11 +295,30 @@ export function updateAdUnitsForAuction(adUnits, floorData, auctionId) {
   });
 }
 
+export function pickRandomModel(modelGroups) {
+  const weightSum = modelGroups.reduce((accum, model) => accum += model.modelWeight, 0);
```

Comment:
Can this weight sum be different each time `pickRandomModel` is called? If not, I'd suggest calculating this once whenever the list of models is set and then reusing the value in this function.

Reply:
Yeo good call. At the time the entire data is set, we can calculate it there and set it as a prop on the object and use that. Will do this change.